dla_cnn/desi/preprocess.py

```diff
@@ -255,5 +258,67 @@ def generate_summary_table(sightlines, output_dir, mode = "w"):
                    "pixel_start_r":sightline.split_point_br,"pixel_end_r":sightline.split_point_rz-1,"wavelength_start_z":10**sightline.loglam[sightline.split_point_rz],\
                    "wavelength_end_z":10**sightline.loglam[-1],"pixel_start_z":sightline.split_point_rz,"pixel_end_z":len(sightline.loglam)-1}
+            
+            dlas_col_density = ""
+            dlas_central_wavelength = ""
+            for dla in sightline.dlas:
+                dlas_col_density += str(dla.col_density)+","
+                dlas_central_wavelength += str(dla.central_wavelength)+","
+            info["dlas_col_density"] = dlas_col_density[:-1]
+            info["dlas_central_wavelength"] = dlas_central_wavelength[:-1]
+            
             #write to the csv file
             summary_table_writer.writerow(info)
+
+def write_summary_table(nums, path, output_path):
+    """
+    Directly read data from fits files and write the summary table, the summary table contains all available sightlines(dlas!=[] and z_qso>2.33) in the given fits files.
+    -----------------------------------------------------------------------------------------------------------------------------------------
+    parameters:
+    nums: list, the given fits files' id, its elements' format is int, and one should make sure all fits files are available before invoking this funciton, otherwise some sightlines can be missed;
+    path: str, the dir of the folder which stores the given fits file, the folder's structure is like folder-fits files' id - fits files , if you are still confused, you can check the below code about read data from the fits file;
+    output_path: str, the dir where the summary table is generated, and if there have been a summary table, then we will remove it and generate a new summary table;
+    ------------------------------------------------------------------------------------------------------------------------------------------
+    retrun:
+    None
+    """
+    #if exists summary table before, remove it
+    if exists(output_path):
+        remove(output_path)
+    def write_as_summary_table(num):
+        """
+        write summary table for a single given fits file, if there have been a summary table then directly write after it, otherwise create a new one
+        ---------------------------------------------------------------------------------------------------------------------------------------------
+        parameter:
+        num: int, the id of the given fits file, e.g. 700
+        ---------------------------------------------------------------------------------------------------------------------------------------------
+        return:
+        None
+        """
+        #read data from fits file
+        file_path = join(path,str(num))
+        spectra = join(file_path,"spectra-16-%s.fits"%num)
```

Comment:
why is the `16` hard-coded?  
More important, what does it mean?

Reply:
That is due to the dataset we use, we use the mock spectrum v9.16, so the fits files' name is in this format, e.g. "spectra-16-700.fits". Maybe we need to set `16` as a parameter.